src/HDF5.jl

```diff
@@ -481,12 +473,9 @@ function h5open(filename::AbstractString, mode::AbstractString = "r"; swmr::Bool
         end
         fid = h5f_open(filename, flag, apl)
     end
-    if close_apl
-        # Close properties manually to avoid errors when the file is
-        # closed before the properties are gc'ed
-        close(apl)
-    end
-    File(fid, filename)
+    close(apl)
+    cpl != DEFAULT_PROPERTIES && close(cpl)
```

Comment:
We were missing this before? I'm guessing this is defensive and the hdf5 library was handling closing this when the file handle was closed previously?

Reply:
Yeah, presumably GC and/or libhdf5 was handling it, but as #754 showed, it's probably advantageous to do this deterministically and early.